### PR TITLE
feat: archive-edit note and rich-text-editor

### PR DESCRIPTION
**Pull request by PuneetP16**
## Rich text Editor Dark: 
image
## Rich text Editor Light:
image
 ## Archived Notes 
image

Most of the code is refactored changes, files to check:
1. NoteEditor.jsx
2. NoteListing.jsx
3. Archive.jsx
4. noteContext.js
5. noteReducer.js

### CODE REVIEW

File: src/components/NoteEditor/NoteEditor.jsx [hunk 1]
```diff
@@ -1,94 +1,105 @@
...
-			setAlert({
-				visibility: true,
-				text: "Enter text before saving the note",
-				type: "alert--danger",
-			});
-		} else {
-			addToNoteList(axiosRequest, noteObj);
+		if (noteObj.title || (bodyText && bodyText !== "<p><br></p>")) {
+			isEditable
+				? updateNote(axiosRequest, { ...noteObj, body: bodyText })
```
Comment (saurabhchirde) [inline src/components/NoteEditor/NoteEditor.jsx:58]: Your note is taking blank spaces as input, You can Trim and then check before adding it to the list.

File: src/components/NoteListWrapper/NoteListWrapper.jsx [hunk 2]
```diff
@@ -1,7 +1,5 @@
 import "./NoteListWrapper.css";
 
-// Type of List: like Pinned, Wrapper Styles like spacing and etc,
-
 export const NoteListWrapper = ({ children }) => {
 	return <section className="note_list_wrapper"> {children} </section>;
 };
```
Comment (saurabhchirde) [inline src/components/NoteListWrapper/NoteListWrapper.jsx:5]: Good to see that you have created a wrapper for notes.

**Review by saurabhchirde — APPROVED** [review]
Screenshot 2022-04-02 at 6 10 14 PM
You can remove Archive, and Delete options from here because, before adding a note, we cannot perform any of the actions. 

Screenshot 2022-04-02 at 6 24 55 PM
If there are no notes then you can hide these names, And just showing those available notes in their particular place would be a good user experience. 
Like, if the user didn't pin any of the notes then you can hide "Pinned Note" and just show all notes without any heading likewise.

Else your code structure is really good and easy to read. 
You can proceed to merge.

File: src/components/NoteListWrapper/NoteListWrapper.jsx (hunk 2 above)
Comment (PuneetP16) [reply src/components/NoteListWrapper/NoteListWrapper.jsx:5]: Thanks for Detailed review and Appreciation.

File: src/components/NoteEditor/NoteEditor.jsx (hunk 1 above)
Comment (PuneetP16) [reply src/components/NoteEditor/NoteEditor.jsx:58]: Will do that.

File: src/App.js [hunk 3]
```diff
@@ -37,11 +37,11 @@ function App() {
 				<Route path="/mockman" element={<MockAPI />} />
 				<Route
 					path="/login"
-					element={isAuth ? <Navigate to="/" replace /> : <Login />}
+					element={isAuth ? <Navigate to="/home" replace /> : <Login />}
```
Comment (m26unkwn) [inline src/App.js:40]: NoteEditor.jsx
NoteListing.jsx
Archive.jsx
noteContext.js
noteReducer.js

File: src/components/NoteEditor/NoteEditor.jsx [hunk 4]
```diff
@@ -1,94 +1,105 @@
...
-				setIsPaletteVisible((w) => !w);
-			}
-		}
+	const noteInputHandler = (e) => {
+		let key = e.target.name;
+		let value = e.target.value;
+		noteDispatch({
+			type: "INPUT_CHANGE",
+			payload: { [key]: value },
+		});
```
Comment (m26unkwn) [inline src/components/NoteEditor/NoteEditor.jsx:41]: This is a good way of using a name for extracting the data..

**Review by m26unkwn — APPROVED** [review]
Your code is clean and readable.
1. Love the improvement of naming conventions.
2. Everything is working well only the UI part.

After doing the suggestion given by @saurabhchirde  you can merge it.

**Review by zoef-yt — APPROVED** [review]
The code looks great. The UI is really amazing. Great work Puneet you are good to merge.

**Comment by PuneetP16** [comment]
Thanks for PR review.

**Pull request merged**